### PR TITLE
Supersearch Post

### PR DESCRIPTION
**Pull request by mastastealth**

### CODE REVIEW

File: _posts/2017-07-18-supersearch-.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+title: Supersearch Me
+categories: development
+tags: development
+author_github: mastastealth
+date:   2017-07-18 15:00:00
+comments: true
+banner_image: supersearch.jpg
+---
+
+As the new Wombat UI evolves and grows, some components take such a large and important role that they are separated out of being a simple UI pattern within our styleguide and turned into a unique, "super" component worthy of it's own sort of development process. "Supersearch" is one of these components.
```
Comment (laurawhalin) [inline _posts/2017-07-18-supersearch-.md]: "it's" in this paragraph should be "its"

File: _posts/2017-07-18-supersearch-.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+On top of the input itself, we wrap the search bar with some helper buttons that tie straight into the search bar's functionality. The idea with Supersearch is that whenever it is used, users recognize the UI pattern of this specific bar and know "oh, this is the search with advanced tag powers to drill down data." The following buttons are currently built-in to the component:
+
+### Predefined Filters
+Supersearch will generally be used in conjunction with the Supertable (another big component, for another post), to help filter the data contained in the table. Wombat tables are quite diverse in the type of data they present, however the Supersearch will always do it's best to provide an initial set of search queries for the user to quickly start hunting down what they need. Usually, these will be the "common" properties someone might search for. For example, if you have a table of campaigns, you might want to quickly search by their status (active, running, done), and the filter will present these to you in a menu so you don't have to type anything just yet.
```
Comment (laurawhalin) [inline _posts/2017-07-18-supersearch-.md]: Second sentence, "it's" should be "its"

File: _posts/2017-07-18-supersearch-.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+<img src="{{ site.url }}/assets/images/ss2.png" style="display:block; margin:0 auto;" />
+
+### Save Queries
+On top of the predefined list, users that find themselves entering the same search tags over and over can choose to (if enabled) save their query into the predefined list for quicker access the next time around!
+
+### Export Data
+If you're a span of spreadsheets, then you will probably enjoying clicking the export button that will allow you to download a file containing all the data of your specific query. This button, like all the reset, are an optional (but handy) piece of the Supersearch component.
```
Comment (laurawhalin) [inline _posts/2017-07-18-supersearch-.md]: Change "enjoying" to "enjoy"

File: _posts/2017-07-18-supersearch-.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## What's Next?
+Supersearch will soon see the light in our future User Management app, so the component will get some real world usage and get adjustments accordingly. A lot of the heavy lifting is done on the API side to return all the smart results based off a user's query, however there are a few plans on enhancing the UI as well. For example, the filter list right now would max out quite quickly at 10-20 or so predefined items before it becomes a "long list." How can this be designed in a way to allow an even greated number of tags? How might this tie into the typing portion of the component? There are some fun design challenges up ahead, so we shall see how this component evolves after it is launched!
```
Comment (laurawhalin) [inline _posts/2017-07-18-supersearch-.md]: change "greated" to "greater"

**Comment by mastastealth** [comment]
@laurawhalin Fixed

**Pull request merged**